### PR TITLE
Plane wave changes

### PR DESCRIPTION
**Pull request by nilsdeppe**
## Proposed changes

- Allow assigning `Variables<Tag0, Tag2>` to `Variables<Tag0, Tag1, Tag2, Tag3,...>` via a `.assign_subset` method
- Make ScalarWave::Solutions::PlaneWave serializable
- Add functions to ScalarWave::Solutions::PlaneWave that allow retrieving a `Variables<Pi, Phi<Dim>, Psi>` and their time derivatives. This is useful for setting initial data from analytic solutions and comparing to them for error measures.
- Make public type aliases between CoordinateMap, CoordinateMapBase, and ElementMap consistent by adding some to the latter two.

### Types of changes:

- [x] Bugfix
- [x] New feature

### Component:

- [x] Code
- [ ] Documentation
- [ ] Build system
- [ ] Continuous integration

### Code review checklist

- [ ] Follows [code review guidelines](https://sxs-collaboration.github.io/spectre/code_review_guide.html)
- [ ] Code has documentation and unit tests
- [ ] Private member variables have a trailing underscore
- [ ] Do not use [Hungarian notation](https://en.wikipedia.org/wiki/Hungarian_notation), e.g. `double* pd_blah` is bad
- [ ] Header order:
  1. hpp corresponding to cpp (only in cpp files)
  2. Blank line (only in cpp files)
  3. STL and externals (in alphabetical order)
  4. Blank line
  5. SpECTRE includes (in alphabetical order)
- [ ] File lists in CMake are alphabetical
- [ ] Correct `noexcept` specification for functions (if unsure, mark `noexcept`)
- [ ] Mark objects `const` whenever possible
- [ ] Almost always `auto`, except with expression templates, i.e. `DataVector`
- [ ] All commits for performance changes provide quantitative evidence and the tests used to obtain said evidence.
- [ ] Make sure error messages are helpful, e.g. "The number of grid points in the matrix 'F' is not the same as the number of grid points in the determinant."
- [ ] Prefix commits addressing PR requests with `fixup`

### CODE REVIEW

File: tests/Unit/DataStructures/Test_Variables.cpp
Comment (osheamonn) [inline tests/Unit/DataStructures/Test_Variables.cpp]: assin-> assign

Comment (nilsdeppe) [reply tests/Unit/DataStructures/Test_Variables.cpp]: Thanks :) Updated!

File: src/DataStructures/Variables.hpp
Comment (wthrowe) [inline src/DataStructures/Variables.hpp]: I think this is the same as `tmpl::list_contains_v<tmpl::list<Tags...>, T>`, but I'm not sure which is actually clearer with the pack expansions, so I'll leave it up to you whether to change it.

Comment (nilsdeppe) [reply src/DataStructures/Variables.hpp]: Actually, by using what you have at the call site (assign_subset) the helper bool can be eliminated completely, which I think is cleanest. I rebased right away because Travis is taking forever right now....

File: src/DataStructures/Variables.hpp
Comment (kidder) [inline src/DataStructures/Variables.hpp]: Tags2 -> SubsetOfTags

File: src/PointwiseFunctions/AnalyticSolutions/WaveEquation/PlaneWave.cpp
Comment (kidder) [inline src/PointwiseFunctions/AnalyticSolutions/WaveEquation/PlaneWave.cpp:88]: This (and the time derivatives) could be done much more efficiently if you didn't call the other member functions

Comment (nilsdeppe) [reply src/PointwiseFunctions/AnalyticSolutions/WaveEquation/PlaneWave.cpp:88]: Yes, but that would lead to a fair bit of code duplication, and I can't see us having computing analytic solutions for scalar waves being something where code duplication is acceptable for increased performance. Thoughts?

**Comment by nilsdeppe** [comment]
I've changed `Tags2` -> `SubsetOfTags`, I haven't heard anything about Larry's other point so I couldn't do anything there.

@wthrowe and @kidder I think this is ready.

[Documentation](http://nilsdeppe.com/spectre/pr_480/)

File: src/PointwiseFunctions/AnalyticSolutions/WaveEquation/PlaneWave.cpp
Comment (kidder) [reply src/PointwiseFunctions/AnalyticSolutions/WaveEquation/PlaneWave.cpp:88]: So I won't hold up this PR for this, but there should be an issue to think about whether we can have an efficient implementation that avoids code duplication

**Review by kidder — APPROVED** [review]

File: src/PointwiseFunctions/AnalyticSolutions/WaveEquation/PlaneWave.cpp
Comment (nilsdeppe) [reply src/PointwiseFunctions/AnalyticSolutions/WaveEquation/PlaneWave.cpp:88]: Reported issue in #489

**Comment by kidder** [comment]
because of long queues for Travis mac builds, I tested this on my laptop

**Pull request merged**
